pkg/controller/container-runtime-config/helpers.go

```diff
@@ -511,15 +513,45 @@ func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec)
 	for i, reg := range imgSpec.RegistrySources.BlockedRegistries {
 		// if there is a match, return all the blocked registries except the one that matched and return an error as well
 		if reg == payloadReg {
-			blockedRegs = append(blockedRegs, imgSpec.RegistrySources.BlockedRegistries[i+1:]...)
-			return blockedRegs, fmt.Errorf("error adding %q to blocked registries, cannot block the registry being used by the payload", payloadReg)
+			// If the payload registry doesn't have mirror rules configured for it, then don't add it to the blocked registries list
+			// TODO maybe? Add an opt-in configuration to ICSP and check if that is set as well
+			if !payloadRegHasMirrorConfig(payloadReg, icspRules, imgSpec.RegistrySources.BlockedRegistries) {
+				blockedRegs = append(blockedRegs, imgSpec.RegistrySources.BlockedRegistries[i+1:]...)
```

Comment:
Assuming the `reg == blockedReg` above becomes a `ScopeIsNestedInsideScope`, it will be possible for more than one `blockedReg` value to be invalid, and in this case each needs to be individually checked and possibly rejected; we can’t terminate on a first violation.

Reply:
fixed